In [1]:
import pandas as pd

## Dataset Load

In [2]:
df = pd.read_csv('pet_health_source_grounded_dataset.csv')

## See The Dataset first 5 rows

In [3]:
df.head()

,case_id,scenario_group_id,pet_type,age,symptom_1,symptom_2,symptom_3,duration,duration_unit,severity,possible_concern,urgency,basic_action,red_flags,record_type,source_name,source_url
0,PH00001,GI01,Dog,0.25,vomiting,diarrhea,reduced appetite,12.0,hours,Moderate,Digestive/Gastrointestinal concern,Moderate,"Keep fresh water available, monitor closely, a...",repeated vomiting; blood in vomit or stool; ma...,source_grounded_derived_scenario,AAHA - Help! Is This a Pet Emergency?,https://www.aaha.org/resources/help-is-this-a-...
1,PH00002,GI01,Dog,0.25,vomiting,diarrhea,reduced appetite,24.0,hours,Moderate,Digestive/Gastrointestinal concern,Moderate,"Keep fresh water available, monitor closely, a...",repeated vomiting; blood in vomit or stool; ma...,source_grounded_derived_scenario,AAHA - Help! Is This a Pet Emergency?,https://www.aaha.org/resources/help-is-this-a-...
2,PH00003,GI01,Dog,0.25,vomiting,diarrhea,reduced appetite,72.0,hours,Moderate,Digestive/Gastrointestinal concern,Moderate,"Keep fresh water available, monitor closely, a...",repeated vomiting; blood in vomit or stool; ma...,source_grounded_derived_scenario,AAHA - Help! Is This a Pet Emergency?,https://www.aaha.org/resources/help-is-this-a-...
3,PH00004,GI01,Dog,0.50,vomiting,diarrhea,reduced appetite,12.0,hours,Moderate,Digestive/Gastrointestinal concern,Moderate,"Keep fresh water available, monitor closely, a...",repeated vomiting; blood in vomit or stool; ma...,source_grounded_derived_scenario,AAHA - Help! Is This a Pet Emergency?,https://www.aaha.org/resources/help-is-this-a-...
4,PH00005,GI01,Dog,0.50,vomiting,diarrhea,reduced appetite,24.0,hours,Moderate,Digestive/Gastrointestinal concern,Moderate,"Keep fresh water available, monitor closely, a...",repeated vomiting; blood in vomit or stool; ma...,source_grounded_derived_scenario,AAHA - Help! Is This a Pet Emergency?,https://www.aaha.org/resources/help-is-this-a-...


## See The Last 5 Rows

In [4]:
df.tail()

,case_id,scenario_group_id,pet_type,age,symptom_1,symptom_2,symptom_3,duration,duration_unit,severity,possible_concern,urgency,basic_action,red_flags,record_type,source_name,source_url
2563,PH02564,LO10,Cat,10.0,mild itching,mild coat thinning,skin flaking,24.0,hours,Mild,Skin/Coat concern,Low,Monitor for progression and arrange routine ve...,severe itching; open wounds; facial swelling; ...,source_grounded_derived_scenario,VCA Animal Hospitals - Pet Skin Care,https://vcahospitals.com/resources/conditions-...
2564,PH02565,LO10,Cat,10.0,mild itching,mild coat thinning,skin flaking,72.0,hours,Mild,Skin/Coat concern,Low,Monitor for progression and arrange routine ve...,severe itching; open wounds; facial swelling; ...,source_grounded_derived_scenario,VCA Animal Hospitals - Pet Skin Care,https://vcahospitals.com/resources/conditions-...
2565,PH02566,LO10,Cat,14.0,mild itching,mild coat thinning,skin flaking,6.0,hours,Mild,Skin/Coat concern,Low,Monitor for progression and arrange routine ve...,severe itching; open wounds; facial swelling; ...,source_grounded_derived_scenario,VCA Animal Hospitals - Pet Skin Care,https://vcahospitals.com/resources/conditions-...
2566,PH02567,LO10,Cat,14.0,mild itching,mild coat thinning,skin flaking,24.0,hours,Mild,Skin/Coat concern,Low,Monitor for progression and arrange routine ve...,severe itching; open wounds; facial swelling; ...,source_grounded_derived_scenario,VCA Animal Hospitals - Pet Skin Care,https://vcahospitals.com/resources/conditions-...
2567,PH02568,LO10,Cat,14.0,mild itching,mild coat thinning,skin flaking,72.0,hours,Mild,Skin/Coat concern,Low,Monitor for progression and arrange routine ve...,severe itching; open wounds; facial swelling; ...,source_grounded_derived_scenario,VCA Animal Hospitals - Pet Skin Care,https://vcahospitals.com/resources/conditions-...


## Dataset Shape

In [5]:
df.shape

(2568, 17)

In this Dataset , There are 2568 rows and 17 columns.

## Dataset Columns

In [6]:
df.columns

Index(['case_id', 'scenario_group_id', 'pet_type', 'age', 'symptom_1',
       'symptom_2', 'symptom_3', 'duration', 'duration_unit', 'severity',
       'possible_concern', 'urgency', 'basic_action', 'red_flags',
       'record_type', 'source_name', 'source_url'],
      dtype='object')

## Dataset Data Type

In [7]:
df.dtypes

case_id               object
scenario_group_id     object
pet_type              object
age                  float64
symptom_1             object
symptom_2             object
symptom_3             object
duration             float64
duration_unit         object
severity              object
possible_concern      object
urgency               object
basic_action          object
red_flags             object
record_type           object
source_name           object
source_url            object
dtype: object

## Dataset Info

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2568 entries, 0 to 2567
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   case_id            2568 non-null   object 
 1   scenario_group_id  2568 non-null   object 
 2   pet_type           2568 non-null   object 
 3   age                2568 non-null   float64
 4   symptom_1          2568 non-null   object 
 5   symptom_2          2568 non-null   object 
 6   symptom_3          2568 non-null   object 
 7   duration           2568 non-null   float64
 8   duration_unit      2568 non-null   object 
 9   severity           2568 non-null   object 
 10  possible_concern   2568 non-null   object 
 11  urgency            2568 non-null   object 
 12  basic_action       2568 non-null   object 
 13  red_flags          2568 non-null   object 
 14  record_type        2568 non-null   object 
 15  source_name        2568 non-null   object 
 16  source_url         2568 

## Statistical Analysis

In [9]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
age,2568.0,4.780666,4.608611,0.25,0.875,3.0,7.75,14.0
duration,2568.0,18.244548,23.410568,0.50,3.000,12.0,24.00,120.0


## Feature Enginnering

# 1. Combine all symptoms into one text column

In [10]:

df["combined_symptoms"] = (
    df["symptom_1"].fillna("") + " " +
    df["symptom_2"].fillna("") + " " +
    df["symptom_3"].fillna("")
).str.strip()